File: jetpack_java/src/main/java/com/kunminx/jetpack_java/sample_03_viewmodel/domain/Request.java


package com.kunminx.jetpack_java.sample_03_viewmodel.domain;

import androidx.lifecycle.LiveData;

import com.kunminx.jetpack_java.common_data.bean.Moment;

import java.util.List;

/**
 * Create by KunMinX at 2020/5/30
 */
public class Request {

    public interface IMomentRequest {

        LiveData<List<Moment>> getListMutableLiveData();

        void requestList();
    }
}

File: jetpack_kotlin/src/main/java/com/example/jetpack_kotlin/sample_03_viewmodel/domain/Request.kt
package com.example.jetpack_kotlin.sample_03_viewmodel.domain

import androidx.lifecycle.LiveData
import com.example.jetpack_kotlin.common_data.bean.Moment

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   22:10
 * description
 */
internal class Request {
    interface IMomentRequest {

        fun getListMutableLiveData(): LiveData<List<Moment>>

        fun requestList()
    }
}